src/app/analyze/stack/stack-details/stack-details.component.ts

```diff
@@ -1,17 +1,26 @@
 import { Stack } from './../../../models/stack';
 import { ComponentAnalysesService } from './../component-analyses.service';
-import { Component, OnInit, Input } from '@angular/core';
+import { Component, OnInit, Input, ViewEncapsulation } from '@angular/core';
 
+import { Logger } from 'ngx-login-client';
 import { StackAnalysesService } from '../stack-analyses.service';
 import { StackAnalysesModel } from '../stack-analyses.model';
 import { Router, ActivatedRoute, Params } from '@angular/router';
+import { RenderComponentService } from './render-component.service';
+import { RenderNextService } from './render-next-service';
+import { AddWorkFlowService } from './add-work-flow.service';
+import { Observable } from 'rxjs/Observable';
+import { FormBuilder } from '@angular/forms';
 
 @Component({
   selector: 'stack-details',
   templateUrl: './stack-details.component.html',
-  styleUrls: ['./stack-details.component.css'],
-  providers: [StackAnalysesService, StackAnalysesModel, ComponentAnalysesService]
+  styleUrls: ['./stack-details.component.scss'],
+  providers: [AddWorkFlowService, RenderNextService, StackAnalysesService,
+    StackAnalysesModel, RenderComponentService, ComponentAnalysesService, Logger],
+  encapsulation: ViewEncapsulation.None
 })
+
```

Comment:
This "component" appears to be a whole app. It should be broken down into consumable components. Monolithic JavaScript files is not the Angular way.